src/components/Form/AutoCompleteAsync.tsx

```diff
@@ -128,14 +144,16 @@ const AutoCompleteAsync = (props: Props) => {
             {!disabled && (
               <ComboboxButton className="absolute inset-y-0 right-0 flex items-center pr-2">
                 <div className="absolute right-0 mr-2 flex items-center text-lg text-secondary-900">
-                  {hasSelection && !loading && !required && (
+                  {shouldShowClearButton && !loading && !required && (
```

Comment:
Let's not change AutoCompleteAsync.

You will notice that issue doesn't persist across all FacilitySelect instances, but rather in ManagePatients and AssetsList; hasSelection normally takes care of this, so look into those two instances and fix those.